src/diff/index.js

```diff
@@ -90,7 +90,7 @@ export function diff(dom, parentDom, newVNode, oldVNode, context, isSvg, excessD
 			let s = c._nextState || c.state;
 			if (newType.getDerivedStateFromProps!=null) {
 				oldState = assign({}, c.state);
-				if (s===c.state) s = assign({}, s);
+				if (s===c.state) s = c._nextState = assign({}, s);
```

Comment:
Awesome :tada: I'd love to have a test case for this so that we don't regress on this feature in any future refactorings :+1:

Reply:
I did it. 👍